pkg/console/operator/operator.go

```diff
@@ -248,55 +259,57 @@ type configSet struct {
 }
 
 func (c *consoleOperator) Sync(ctx context.Context, controllerContext factory.SyncContext) error {
-	operatorConfig, err := c.operatorConfigClient.Get(ctx, api.ConfigResourceName, metav1.GetOptions{})
+	operatorConfig, err := c.consoleOperatorLister.Get(api.ConfigResourceName)
 	if err != nil {
 		klog.Error("failed to retrieve operator config: %v", err)
 		return err
 	}
 
 	startTime := time.Now()
 	klog.V(4).Infof("started syncing operator %q (%v)", operatorConfig.Name, startTime)
-	defer klog.V(4).Infof("finished syncing operator %q (%v)", operatorConfig.Name, time.Since(startTime))
+	defer func() {
+		klog.V(4).Infof("finished syncing operator %q (%v)", operatorConfig.Name, time.Since(startTime))
+	}()
 
 	// ensure we have top level console config
-	consoleConfig, err := c.consoleConfigClient.Get(ctx, api.ConfigResourceName, metav1.GetOptions{})
+	consoleConfig, err := c.consoleConfigLister.Get(api.ConfigResourceName)
 	if err != nil {
 		klog.Errorf("console config error: %v", err)
 		return err
 	}
 
 	// we need infrastructure config for apiServerURL
-	infrastructureConfig, err := c.infrastructureConfigClient.Get(ctx, api.ConfigResourceName, metav1.GetOptions{})
+	infrastructureConfig, err := c.infrastructureLister.Get(api.ConfigResourceName)
 	if err != nil {
 		klog.Errorf("infrastructure config error: %v", err)
 		return err
 	}
 
-	proxyConfig, err := c.proxyConfigClient.Get(ctx, api.ConfigResourceName, metav1.GetOptions{})
+	proxyConfig, err := c.proxyConfigLister.Get(api.ConfigResourceName)
 	if err != nil {
 		klog.Errorf("proxy config error: %v", err)
 		return err
 	}
 
-	oauthConfig, err := c.oauthConfigClient.Get(ctx, api.ConfigResourceName, metav1.GetOptions{})
+	oauthConfig, err := c.oauthConfigLister.Get(api.ConfigResourceName)
 	if err != nil {
 		klog.Errorf("oauth config error: %v", err)
 		return err
 	}
 
-	ingressConfig, err := c.ingressConfigClient.Get(ctx, api.ConfigResourceName, metav1.GetOptions{})
+	ingressConfig, err := c.ingressConfigLister.Get(api.ConfigResourceName)
 	if err != nil {
 		klog.Errorf("ingress config error: %v", err)
 		return err
 	}
 
 	configs := configSet{
-		Console:        consoleConfig,
-		Operator:       operatorConfig,
-		Infrastructure: infrastructureConfig,
-		Proxy:          proxyConfig,
-		OAuth:          oauthConfig,
-		Ingress:        ingressConfig,
+		Console:        consoleConfig.DeepCopy(),
+		Operator:       operatorConfig.DeepCopy(),
+		Infrastructure: infrastructureConfig.DeepCopy(),
+		Proxy:          proxyConfig.DeepCopy(),
+		OAuth:          oauthConfig.DeepCopy(),
+		Ingress:        ingressConfig.DeepCopy(),
```

Comment:
See https://github.com/openshift/console-operator/pull/814#discussion_r1411768006.

I suppose some of them could be just the original object as they would only ever be read but since I don't know the code too well, it's easier for me to just create copies right away.

Reply:
Another point I wanted to raise - this actually adheres to the original workflow of the operator but if anyone's addressing the tech debt later, I would strongly discourage this kind of pre-sync state store in favour of just-in-time state retrievals.

This way you would get rid of any possible side-effects which is a typical problem in these situations.